File: src/mysqlcookbook/recipes/api/01_connect/connect.pl
#!/usr/bin/perl
# connect.pl: connect to the MySQL server

use strict;
use warnings;
use DBI;

my $dsn = "DBI:mysql:host=localhost;database=cookbook";
#@ _CONNECT_
my $dbh = DBI->connect ($dsn, "cbuser", "cbpass")
            or die "Cannot connect to server\n";
#@ _CONNECT_
print "Connected\n";
$dbh->disconnect ();
print "Disconnected\n";

File: src/mysqlcookbook/recipes/api/01_connect/connect.php
<?php
# connect.php: connect to the MySQL server

#@ _FRAG_
#@ _FRAG2_
try
{
  $dsn = "mysql:host=localhost;dbname=cookbook";
  $dbh = new PDO ($dsn, "cbuser", "cbpass");
  print ("Connected\n");
}
catch (PDOException $e)
{
  die ("Cannot connect to server\n");
}
#@ _FRAG2_
$dbh = NULL;
print ("Disconnected\n");
#@ _FRAG_
?>
